be/src/storage/lake/transactions.cpp

```diff
@@ -53,9 +71,37 @@ static void clear_remote_snapshot_async(TabletManager* tablet_mgr, int64_t table
     files_to_delete->emplace_back(std::move(slog_path));
 }
 
+void adjust_base_version(int64_t tablet_id, TabletManager* tablet_mgr, int64_t* base_version) {
+    int64_t version = *base_version;
+    auto metadata = tablet_mgr->get_latest_cached_tablet_metadata(tablet_id);
+    if (metadata != nullptr) {
+        version = std::max(version, metadata->version());
+    }
+
+    auto index_version = tablet_mgr->update_mgr()->get_primary_index_data_version(tablet_id);
+    if (index_version > version) {
+        auto res = tablet_mgr->get_tablet_metadata(tablet_id, index_version);
+        if (res.ok()) {
+            version = std::max(version, index_version);
```

Comment:
```suggestion
            version = index_version;
```
You've already checked above that index_version > version